atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -651,13 +675,161 @@ def set_build_image(self):
             raise RuntimeError("Build object is malformed, failed to fetch buildroot image")
 
         if build_kind == 'DockerImage':
-            self.config_kwargs['build_image'] = build_name
+            return build_name
         else:
             raise RuntimeError("Build kind isn't 'DockerImage' but %s" % build_kind)
 
+    def process_image_from(self, image_from):
+        build_image = None
+        imagestream = None
+
+        if image_from['kind'] == 'DockerImage':
+            build_image = image_from['name']
+        elif image_from['kind'] == 'ImageStreamTag':
+            imagestream = image_from['name']
+        else:
+            raise UnknownKindException
+
+        return build_image, imagestream
+
+    def set_build_image(self, platforms):
+        """
+        Overrides build_image for worker, to be same as in orchestrator build
+        """
+        current_platform = platform.processor()
+        orchestrator_platform = current_platform or 'x86_64'
+
+        build_config_name = None
+        build_image = None
+        imagestream = None
+        current_buildimage = self.get_current_buildimage()
+        status = get_build_json().get("status", {})
+
+        # orchestrator platform is same as platform on which we want to built on,
+        # so we can use the same image
+        if platforms == set([orchestrator_platform]):
+            self.build_image_digests[orchestrator_platform] = current_buildimage
+            return
+
+        # BuildConfig exists
+        if 'config' in status:
+            config = status['config']
+            if config['kind'] == 'BuildConfig':
+                build_config_name = config['name']
+            else:
+                raise RuntimeError("Build config type isn't BuildConfig : %s" % config['kind'])
+
+        # get image build from build metadata, which is set for direct builds
+        # this is explicitly set by osbs-client, it isn't default OpenShift behaviour
+        else:
+            annotations = get_build_json().get("metadata", {}).get('annotations', {})
+            if 'from' in annotations:
+                scratch_from = json.loads(annotations['from'])
+
+                try:
+                    build_image, imagestream = self.process_image_from(scratch_from)
+                except UnknownKindException:
+                    raise RuntimeError("Build annotation has unknown 'kind' %s" %
+                                       scratch_from['kind'])
+            else:
+                raise RuntimeError("Build wasn't created from BuildConfig and neither"
+                                   " has 'from' annotation, which is needed for specified arch")
+
+        osbs = get_openshift_session(self.workflow, self.openshift_fallback)
```

Comment:
can this be moved up to be above the check for config in status? it doesn't look like it depends on the results of the if else block.

Reply:
Could turn into a property so it's only initialized when needed, similar to how `resolve_composes` plugin handles `odcs_client` and `koji_client`.

Reply:
at this point we certainly need that session, in other cases it would raise and we wouldn't be at this point

Reply:
ok will make it property